src/ja/server/dispatcher/dispatcher.py

```diff
@@ -36,17 +38,35 @@ def set_distribution(self, job_distribution: JobDistribution) -> None:
                 proxy = self._proxy_factory.get_proxy(work_machine)
                 if job.status == JobStatus.RUNNING:
                     if previous_status is None or previous_status == JobStatus.QUEUED:
-                        proxy.dispatch_job(job)
+                        try:
+                            proxy.dispatch_job(job)
+                        except Exception:
+                            self._timeout(job_entry)
                     elif previous_status == JobStatus.PAUSED:
-                        proxy.resume_job(job.uid)
+                        try:
+                            proxy.resume_job(job.uid)
+                        except Exception:
+                            self._timeout(job_entry)
                     new_statuses[job.uid] = job.status
                 elif job.status == JobStatus.CANCELLED:
-                    proxy.cancel_job(job.uid)  # Do not add job back to self._previous_statuses
+                    try:
+                        proxy.cancel_job(job.uid)  # Do not add job back to self._previous_statuses
+                    except Exception:
+                        self._timeout(job_entry)
                 elif job.status == JobStatus.PAUSED:
-                    proxy.pause_job(job.uid)
+                    try:
+                        proxy.pause_job(job.uid)
+                    except Exception:
+                        self._timeout(job_entry)
                     new_statuses[job.uid] = job.status
                 elif job.status == JobStatus.QUEUED:
                     new_statuses[job.uid] = job.status
                 else:
                     raise ValueError("Received unexpected state %s for job with UID %s." % (job.status.name, job.uid))
         self._previous_statuses = new_statuses
+
+    def _timeout(self, entry: DatabaseJobEntry) -> None:
+        work_machine = entry.assigned_machine
+        print("Lost connection to Work machine with id %s: " % work_machine.uid)
```

Comment:
Would make sense to use the new logging methods, the loggin PR has just been merged.